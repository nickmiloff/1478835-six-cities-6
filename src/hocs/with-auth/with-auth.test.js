import React from 'react';
import PropTypes from 'prop-types';
import {render} from '@testing-library/react';
import {TestStore as Test} from '../../components/test/test';

import withAuth from './with-auth';

describe(`withAuth HOC test`, () => {
  const MockComponent = ({isAuth}) => (<span>{isAuth.toString()}</span>);
  const MockComponentWithHOC = withAuth(MockComponent);

  MockComponent.propTypes = {isAuth: PropTypes.bool.isRequired};

  it(`'withAuth HOC' should render correctly with authorized user`, () => {
    const {container} = render(
        <Test initial={false}>
          <MockComponentWithHOC />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });

  it(`'withAuth HOC' should render correctly with unauthorized user`, () => {
    const {container} = render(
        <Test>
          <MockComponentWithHOC />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });
});
