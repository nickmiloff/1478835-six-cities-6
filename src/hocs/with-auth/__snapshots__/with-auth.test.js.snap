// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuth HOC test 'withAuth HOC' should render correctly with authorized user 1`] = `
<div>
  <span>
    true
  </span>
</div>
`;

exports[`withAuth HOC test 'withAuth HOC' should render correctly with unauthorized user 1`] = `
<div>
  <span>
    false
  </span>
</div>
`;
